src/stdlib_io_npy_load.fypp

```diff
@@ -0,0 +1,533 @@
+! SPDX-Identifier: MIT
+
+#:include "common.fypp"
+#:set RANKS = range(1, MAXRANK + 1)
+#:set KINDS_TYPES = REAL_KINDS_TYPES + INT_KINDS_TYPES + CMPLX_KINDS_TYPES
+
+!> Implementation of loading npy files into multidimensional arrays
+submodule (stdlib_io_npy) stdlib_io_npy_load
+    use stdlib_error, only : error_stop
+    use stdlib_strings, only : to_string, starts_with
+    implicit none
+
+contains
+
+#:for k1, t1 in KINDS_TYPES
+  #:for rank in RANKS
+    !> Load a ${rank}$-dimensional array from a npy file
+    module subroutine load_npy_${t1[0]}$${k1}$_${rank}$(filename, array, iostat, iomsg)
+        !> Name of the npy file to load from
+        character(len=*), intent(in) :: filename
+        !> Array to be loaded from the npy file
+        ${t1}$, allocatable, intent(out) :: array${ranksuffix(rank)}$
+        !> Error status of loading, zero on success
+        integer, intent(out), optional :: iostat
+        !> Associated error message in case of non-zero status code
+        character(len=:), allocatable, intent(out), optional :: iomsg
+
+        character(len=*), parameter :: vtype = type_${t1[0]}$${k1}$
+        integer, parameter :: rank = ${rank}$
+
+        integer :: io, stat
+        character(len=:), allocatable :: msg
+
+        open(newunit=io, file=filename, form="unformatted", access="stream", iostat=stat)
```

Comment:
The block allows to jump to the error handling in case of non-zero exit status, the alternative is using `if (stat == 0)` for all commands or introducing a goto.

Reply:
Sorry. I missed this goal at the first read. Well done!